From 097221f7026cbda26d77faf51ac8d645f58aa94f Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Thu, 10 Aug 2023 17:18:54 +0200
Subject: [PATCH 341/954] feat(psci): export psci_set_target_local_pwr_states()

After a standby on STM32MP2, the content of SYSRAM (where BL31 is loaded)
is lost. We should then have a mean to restore the power state after
coming back from low-power state. This is done by making the function
psci_set_target_local_pwr_states() usable outside PSCI files.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git abb2165303f619e69795ab0323fba5e44acb1c59 feat(psci): export psci_set_target_local_pwr_states()]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: I6a14de0184cde98640a4e8b272da8e2413b666de
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/327282
Tested-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 include/lib/psci/psci.h | 4 +++-
 lib/psci/psci_private.h | 2 --
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/lib/psci/psci.h b/include/lib/psci/psci.h
index 7381fcdc98..24b492695e 100644
--- a/include/lib/psci/psci.h
+++ b/include/lib/psci/psci.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2013-2019, Arm Limited and Contributors. All rights reserved.
+ * Copyright (c) 2013-2023, Arm Limited and Contributors. All rights reserved.
  * Copyright (c) 2023, NVIDIA Corporation. All rights reserved.
  *
  * SPDX-License-Identifier: BSD-3-Clause
@@ -387,6 +387,8 @@ int psci_set_suspend_mode(unsigned int mode);
 #endif
 void __dead2 psci_power_down_wfi(void);
 void psci_arch_setup(void);
+void psci_set_target_local_pwr_states(unsigned int end_pwrlvl,
+				      const psci_power_state_t *target_state);
 
 #endif /*__ASSEMBLER__*/
 
diff --git a/lib/psci/psci_private.h b/lib/psci/psci_private.h
index 2eb4a9b705..c7cc47eda4 100644
--- a/lib/psci/psci_private.h
+++ b/lib/psci/psci_private.h
@@ -297,8 +297,6 @@ void psci_restore_req_local_pwr_states(unsigned int cpu_idx,
 #endif
 void psci_get_target_local_pwr_states(unsigned int end_pwrlvl,
 				      psci_power_state_t *target_state);
-void psci_set_target_local_pwr_states(unsigned int end_pwrlvl,
-				      const psci_power_state_t *target_state);
 int psci_validate_entry_point(entry_point_info_t *ep,
 			uintptr_t entrypoint, u_register_t context_id);
 void psci_get_parent_pwr_domain_nodes(unsigned int cpu_idx,
-- 
2.53.0

